### ISSUE TITLE
AutoConsumeSchema NPE

### ISSUE THREAD

**Issue by TIBCOeddie** [apache/pulsar#9806]
**Describe the bug**
NPE seen by the client when it tried to use AUTO_CONSUME against a topic

> (fragment deploy): 2021-03-03 21:28:38.154000-0500 [15819:OperatorThread(default.Java:1)] ERROR pulsar.PulsarConsumer.default.Java: Fatal error
> (fragment deploy): java.lang.NullPointerException: null
> (fragment deploy): at org.apache.pulsar.client.impl.schema.AutoConsumeSchema.generateSchema(AutoConsumeSchema.java:154)
> (fragment deploy): at org.apache.pulsar.client.impl.schema.AutoConsumeSchema.decode(AutoConsumeSchema.java:91)
> (fragment deploy): at org.apache.pulsar.client.impl.schema.AutoConsumeSchema.decode(AutoConsumeSchema.java:40)
> (fragment deploy): at org.apache.pulsar.client.impl.MessageImpl.getValue(MessageImpl.java:301)

**To Reproduce**
Consumer created this way:
```
pulsarConsumer = pulsarClient.newConsumer(org.apache.pulsar.client.api.Schema.AUTO_CONSUME())
					.topic(getTopic())
					.subscriptionName(UUID.randomUUID().toString())
					.subscribe();
...
Message<GenericRecord> message = pulsarConsumer.receive(5, TimeUnit.SECONDS);
if (message == null) continue;
GenericRecord record = message.getValue(); <-- NPE inside
```

Message sent to server as follows:
`./pulsar-client produce topic1 -m "hello"`
which, understandably is not JSON nor AVRO but an NPE should have been avoided

**Expected behavior**
No NPE. If there's some configuration that could have been done better, the error should identify it.

**Additional context**
Java client 2.7.0

**Comment by codelipenghui** [comment]
It should not throw NPE, the auto_consume schema only works for the struct schema such as Avro, Json. I labeled as release/2.7.2 to make sure this one can be fixed in the next release.

**Comment by eolivelli** [comment]
This issue has been fixed in 2.8.0. we enhanced a lot the support for AutoConsumeSchema, now it is able to deal with every Schema and with topics without a Schema.

Look for GenericObject related issues and PRs
